pkg/exchange/ftx/exchange.go

```diff
@@ -142,7 +142,51 @@ func (e *Exchange) QueryAccountBalances(ctx context.Context) (types.BalanceMap,
 }
 
 func (e *Exchange) QueryKLines(ctx context.Context, symbol string, interval types.Interval, options types.KLineQueryOptions) ([]types.KLine, error) {
-	panic("implement me")
+	var since, until time.Time
+	if options.StartTime != nil {
+		since = *options.StartTime
+	}
+	if options.EndTime != nil {
+		until = *options.EndTime
+	} else {
+		until = time.Now()
+	}
+	if since.After(until) {
+		return nil, fmt.Errorf("invalid query klines time range, since: %+v, until: %+v", since, until)
+	}
+	if !isIntervalSupportedInKLine(interval) {
+		return nil, fmt.Errorf("interval %s is not supported", interval.String())
+	}
+	resp, err := e.newRest().HistoricalPrices(ctx, symbol, interval, int64(options.Limit), since, until)
+	if err != nil {
+		return nil, err
+	}
+	if !resp.Success {
+		return nil, fmt.Errorf("ftx returns failure")
+	}
+
+	var kline []types.KLine
+	for _, r := range resp.Result {
+		globalKline, err := toGlobalKLine(symbol, interval, r)
+		if err != nil {
+			return nil, err
+		}
+		kline = append(kline, globalKline)
+	}
+	return kline, nil
+}
+
+func isIntervalSupportedInKLine(interval types.Interval) bool {
+	_, ok := map[int]struct{}{
+		15:    {},
```

Comment:
we need at least 1m and 5m for back testing

Reply:
Yes it's supported by ftx.